src/GenericConsole.php

```diff
@@ -5,43 +5,55 @@
 namespace Tempest\Console;
 
 use Closure;
-use Tempest\Console\Actions\RenderConsoleComponent;
+use Tempest\Console\Components\ComponentRenderer;
 use Tempest\Console\Components\ConfirmComponent;
 use Tempest\Console\Components\MultipleChoiceComponent;
 use Tempest\Console\Components\PasswordComponent;
 use Tempest\Console\Components\ProgressBarComponent;
 use Tempest\Console\Components\QuestionComponent;
 use Tempest\Console\Components\SearchComponent;
 use Tempest\Console\Components\TextBoxComponent;
+use Tempest\Console\Highlight\TempestConsoleLanguage;
+use Tempest\Console\Input\InputBuffer;
+use Tempest\Console\Output\OutputBuffer;
+use Tempest\Highlight\Highlighter;
 
-final readonly class GenericConsole implements Console
+final class GenericConsole implements Console
 {
+    private ?string $label = null;
+
     public function __construct(
-        private ConsoleInput $input,
-        private ConsoleOutput $output,
+        private readonly OutputBuffer $output,
```

Comment:
The biggest refactor, we now have `OutputBuffer`, `InputBuffer`, `ComponentRenderer` and `Highlighter` as dependencies. Each of them can be mocked/replaced depending on the context.

The end goal of this PR was to make is super trivial to configure the console so that it

- Writes to StdOut when running within the terminal
- Writes to a log file when running as a background task
- Disallows interactive commands that require user input, which currently isn't supported in tests (wip), and won't be supported in background jobs

Reply:
Oh man! I like this *a lot* 🎉 

It's clean and will make testing a breeze